skbio/diversity/tests/test_driver.py

```diff
@@ -0,0 +1,590 @@
+                     u'0.75,OTU2:0.75):1.25):0.0)root;'))
+        counts = [1, 2, 3]
+        otu_ids = ['OTU1', 'OTU2']
+        self.assertRaises(ValueError, beta_diversity,
+                          'weighted_unifrac', counts, otu_ids=otu_ids, tree=t)
+        self.assertRaises(ValueError, beta_diversity,
+                          'unweighted_unifrac', counts, otu_ids=otu_ids,
+                          tree=t)
+        t = TreeNode.read(
+            StringIO(u'(((((OTU1:0.5,OTU2:0.5):0.5,OTU3:1.0):1.0):0.0,(OTU4:'
+                     u'0.75,OTU2:0.75):1.25):0.0)root;'))
+        counts = [1, 2]
+        otu_ids = ['OTU1', 'OTU2', 'OTU3']
+        self.assertRaises(ValueError, beta_diversity,
+                          'weighted_unifrac', counts, otu_ids=otu_ids, tree=t)
+        self.assertRaises(ValueError, beta_diversity,
+                          'unweighted_unifrac', counts, otu_ids=otu_ids,
+                          tree=t)
+
+        # tree with no branch lengths
+        t = TreeNode.read(
+            StringIO(u'((((OTU1,OTU2),OTU3)),(OTU4,OTU5));'))
+        counts = [1, 2, 3]
+        otu_ids = ['OTU1', 'OTU2', 'OTU3']
+        self.assertRaises(ValueError, beta_diversity,
+                          'weighted_unifrac', counts, otu_ids=otu_ids, tree=t)
+        self.assertRaises(ValueError, beta_diversity,
+                          'unweighted_unifrac', counts, otu_ids=otu_ids,
+                          tree=t)
+
+        # tree missing some branch lengths
+        t = TreeNode.read(
+            StringIO(u'(((((OTU1,OTU2:0.5):0.5,OTU3:1.0):1.0):0.0,(OTU4:'
+                     u'0.75,OTU5:0.75):1.25):0.0)root;'))
+        counts = [1, 2, 3]
+        otu_ids = ['OTU1', 'OTU2', 'OTU3']
+        self.assertRaises(ValueError, beta_diversity,
+                          'weighted_unifrac', counts, otu_ids=otu_ids, tree=t)
+        self.assertRaises(ValueError, beta_diversity,
+                          'unweighted_unifrac', counts, otu_ids=otu_ids,
+                          tree=t)
+
+        # some otu_ids not present in tree
+        t = TreeNode.read(
+            StringIO(u'(((((OTU1:0.5,OTU2:0.5):0.5,OTU3:1.0):1.0):0.0,(OTU4:'
+                     u'0.75,OTU5:0.75):1.25):0.0)root;'))
+        counts = [1, 2, 3]
+        otu_ids = ['OTU1', 'OTU2', 'OTU42']
+        self.assertRaises(MissingNodeError, beta_diversity,
+                          'weighted_unifrac', counts, otu_ids=otu_ids, tree=t)
+        self.assertRaises(MissingNodeError, beta_diversity,
+                          'unweighted_unifrac', counts, otu_ids=otu_ids,
+                          tree=t)
+
+    def test_empty(self):
+        # array of empty vectors
+        actual = beta_diversity('euclidean',
+                                np.array([[], []], dtype=np.int64),
+                                ids=['a', 'b'])
+        expected_dm = DistanceMatrix([[0.0, 0.0], [0.0, 0.0]], ['a', 'b'])
+        npt.assert_array_equal(actual, expected_dm)
```

Comment:
For these `beta_diversity` tests, use `self.assertEqual(actual, expected_dm)`, that will correctly compare `DistanceMatrix` objects for equality.